pkg/server/home_workspaces.go

```diff
@@ -0,0 +1,664 @@
+			return
+		}
+		if decision != authorizer.DecisionAllow {
+			responsewriters.Forbidden(ctx, attributes, rw, req, reason, homeWorkspaceCodecs)
+			return
+		}
+
+		// TODO: The way we build this URL here is not fully compatible with sharding long-term:
+		// - short term (what is done now): use genericConfig.ExternalAddress
+		// - medium term: shards will know their name eventually, and with that we can lookup the external URL of the shard
+		// - long-term: create home workspace already on `~` access in order to make scheduling happen and then we know the real external URL.
+		homeWorkspaceURL := &url.URL{Scheme: "https", Host: h.externalHost, Path: homeLogicalClusterName.Path()}
+
+		parent, name := homeLogicalClusterName.Split()
+		homeWorkspace := &tenancyv1beta1.Workspace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:        name,
+				ClusterName: parent.String(),
+			},
+			Spec: tenancyv1beta1.WorkspaceSpec{
+				Type: tenancyv1alpha1.ClusterWorkspaceTypeReference{
+					Name: tenancyv1alpha1.ClusterWorkspaceTypeName("Home"),
+					Path: tenancyv1alpha1.RootCluster.String(),
+				},
+			},
+			Status: tenancyv1beta1.WorkspaceStatus{
+				URL: homeWorkspaceURL.String(),
+			},
+		}
+
+		responsewriters.WriteObjectNegotiated(homeWorkspaceCodecs, negotiation.DefaultEndpointRestrictions, tenancyv1beta1.SchemeGroupVersion, rw, req, http.StatusOK, homeWorkspace)
+		return
+	}
+
+	needsAutomaticCreation, workspaceType := h.needsAutomaticCreation(lcluster.Name)
+	if !needsAutomaticCreation {
+		h.apiHandler.ServeHTTP(rw, req)
+		return
+	}
+
+	isHome := workspaceType == "Home"
+	if foundLocally, retryAfterSeconds, err := h.searchForReadyWorkspaceInLocalInformers(lcluster.Name, isHome, user.GetName()); foundLocally {
+		if retryAfterSeconds > 0 {
+			// Return a 429 status asking the client to try again after the creationDelay
+			rw.Header().Set("Retry-After", fmt.Sprintf("%d", retryAfterSeconds))
+			http.Error(rw, "Creating the home workspace", http.StatusTooManyRequests)
+			return
+		}
+		h.apiHandler.ServeHTTP(rw, req)
+		return
+	} else if err != nil {
+		responsewriters.InternalError(rw, req, err)
+		return
+	}
+
+	// Home or bucket workspace not found in the local informer
+	// Let's try to create it
+
+	// But first check we have the right to do so.
+	attributes := homeWorkspaceAuthorizerAttributes(user, "create")
+
+	if isHome && lcluster.Name != h.getHomeLogicalClusterName(user.GetName()) {
+		// If we're checking a home workspace or home bucket workspace, but not of the consistent user, let's refuse.
+		klog.Errorf("failed to authorize user %q to create a home workspace %q: home workspace can only be created by the user of the home workspace", user.GetName(), lcluster.Name)
+		responsewriters.Forbidden(ctx, attributes, rw, req, authorization.WorkspaceAcccessNotPermittedReason, homeWorkspaceCodecs)
+		return
+	}
+
+	// Test if the user has the right to create his Home workspace when it doesn't exists
```

Comment:
When is this not true ?

Reply:
see answer to comment https://github.com/kcp-dev/kcp/pull/1373/files/0faa774d210b4af08718c56528ea795db4223091#r916265720